flipt-java/src/main/java/io/flipt/api/flags/Flag.java

```diff
@@ -0,0 +1,200 @@
+  private final Map<String, String> headers;
+  private final ObjectMapper objectMapper;
+  private final Consumer<FliptException> unhandledExceptionProcessor;
+
+  private Flag(FlagBuilder builder) {
+    this.httpClient = builder.httpClient;
+    this.baseURL = builder.baseURL;
+    this.authenticationStrategy = builder.authenticationStrategy;
+    this.headers = builder.headers;
+    this.objectMapper =
+        JsonMapper.builder()
+            .addModule(new Jdk8Module())
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+            .addModule(new JavaTimeModule())
+            .build();
+    this.unhandledExceptionProcessor = builder.unhandledExceptionProcessor;
+  }
+
+  public static FlagBuilder builder() {
+    return new FlagBuilder();
+  }
+
+  public static class FlagBuilder {
+    private OkHttpClient httpClient;
+    private String baseURL;
+    private AuthenticationStrategy authenticationStrategy;
+    private Map<String, String> headers;
+    private Consumer<FliptException> unhandledExceptionProcessor;
+
+    public FlagBuilder httpClient(OkHttpClient httpClient) {
+      this.httpClient = httpClient;
+      return this;
+    }
+
+    public FlagBuilder baseURL(String baseURL) {
+      this.baseURL = baseURL;
+      return this;
+    }
+
+    public FlagBuilder authenticationStrategy(AuthenticationStrategy authenticationStrategy) {
+      this.authenticationStrategy = authenticationStrategy;
+      return this;
+    }
+
+    public FlagBuilder headers(Map<String, String> headers) {
+      this.headers = headers;
+      return this;
+    }
+
+    public FlagBuilder setUnhandledExceptionProcessor(Consumer<FliptException> processor) {
+      this.unhandledExceptionProcessor = processor;
+      return this;
+    }
+
+    public Flag build() {
+      return new Flag(this);
+    }
+  }
+
+  /**
+   * Get a flag by namespace key and flag key
+   *
+   * @param namespaceKey the {@link String}
+   * @param flagKey the {@link String}
+   * @return a {@link io.flipt.api.flags.models.Flag} containing the flag information
+   * @throws FlagException if error occurs during the process, such as a network issue or invalid
+   *     request
+   */
+  public io.flipt.api.flags.models.Flag getFlag(String namespaceKey, String flagKey)
+      throws FlagException {
+    return getFlag(namespaceKey, flagKey, CommonParameters.builder().build());
+  }
+
+  /**
+   * Get a flag by namespace key and flag key
+   *
+   * @param namespaceKey the {@link String}
+   * @param flagKey the {@link String}
+   * @param params the {@link CommonParameters}
+   * @return a {@link io.flipt.api.flags.models.Flag} containing the flag information
+   * @throws FlagException if error occurs during the process, such as a network issue or invalid
+   *     request
+   */
+  public io.flipt.api.flags.models.Flag getFlag(
+      String namespaceKey, String flagKey, CommonParameters params) throws FlagException {
+    return this.makeGetRequest(
```

Comment:
we should probably set the namespace to `default` if empty here and below. I'll create an issue to do the same in the Python SDK

Reply:
#357